src/qilib/configuration_helper/instrument_adapter_factory.py

```diff
@@ -58,15 +59,21 @@ def get_instrument_adapter(cls, instrument_adapter_class_name: str, address: str
 
         Returns:
              An instance of the requested InstrumentAdapter.
+
+        Raises:
+            ValueError: If adapter is not found or error occurred while importing the adapter.
+
         """
         instrument_adapter_key = instrument_adapter_class_name, str(address)
         if instrument_adapter_key in cls.instrument_adapters:
             return cls.instrument_adapters[instrument_adapter_key]
+        if cls.is_instrument_adapter(instrument_adapter_class_name):
+            adapter = cast(InstrumentAdapter,
+                           vars(qilib.configuration_helper.adapters)[instrument_adapter_class_name](address))
+        elif instrument_adapter_class_name in cls.failed_adapters:
```

Comment:
Shouldn't we check first whether the instrument_adapter is in the failed_adapters? Does it make sense to potentially get the adapter address, if we already determined that the import has failed

Reply:
I think you should get a working one if you can, but good point, I'll have a look and a thought.

Reply:
done